### PR TITLE
Compact high-traffic MAP skill playbooks

### PR DESCRIPTION
**Pull request by azalio**
## Summary
- Compact /map-plan, /map-efficient, /map-check, and /map-review active SKILL.md bodies under 500 lines
- Move low-frequency examples, troubleshooting, rationale, command matrices, wave details, and section rubrics into bundled supporting files
- Add regression coverage, docs, improvement-plan bookkeeping, and learned rules for high-traffic skill compactness

## User impact
The main MAP golden-path skills now carry less recurring context after invocation and compaction while preserving mandatory execution gates and full supporting references for edge cases.

## Validation
- pytest tests/test_skills.py tests/test_template_sync.py -v
- make lint
- pytest -m "not slow"
- uv run --no-sync mapify init <temp> --no-git --mcp none, then inspected installed compact skills and supporting refs
- PYTHONPATH=src python -m mapify_cli.skill_ir src/mapify_cli/templates/skills src/mapify_cli/templates/codex/skills --format json

### CODE REVIEW

File: src/mapify_cli/templates/skills/map-review/SKILL.md [hunk 1]
```diff
@@ -196,336 +134,101 @@ git diff HEAD
...
-# concern, not a per-run concern).
-DETACHED_RESULT=$(python3 .map/scripts/map_step_runner.py prepare_detached_review "$BUNDLE_JSON_PATH")
-DETACHED_STATUS=$(echo "$DETACHED_RESULT" | python3 -c "import sys,json; d=json.load(sys.stdin); print(d['status'])")
-DETACHED_PATH=$(echo "$DETACHED_RESULT" | python3 -c "import sys,json; d=json.load(sys.stdin); print(d.get('worktree_path') or '')")
-DETACHED_REASON=$(echo "$DETACHED_RESULT" | python3 -c "import sys,json; d=json.load(sys.stdin); print(d.get('reason') or '')")
+DETACHED_PATH=""
+if [ "$DETACHED_FLAG" = "true" ]; then
+  # EC-15: prepare detached review once; compare runs reuse the same path.
+  DETACHED_JSON=$(python3 .map/scripts/map_step_runner.py prepare_detached_review)
+  DETACHED_PATH=$(echo "$DETACHED_JSON" | python3 -c "import sys,json; print(json.load(sys.stdin).get('detached_path',''))")
```
[the remark this answers is not in the document]
Comment (azalio) [reply src/mapify_cli/templates/skills/map-review/SKILL.md]: Fixed in a22ad14: the detached helper snippet now parses status, reason, and worktree_path from prepare_detached_review and passes BUNDLE_JSON_PATH.

File: .claude/skills/map-review/SKILL.md [hunk 2]
```diff
@@ -196,336 +134,101 @@ git diff HEAD
...
-# concern, not a per-run concern).
-DETACHED_RESULT=$(python3 .map/scripts/map_step_runner.py prepare_detached_review "$BUNDLE_JSON_PATH")
-DETACHED_STATUS=$(echo "$DETACHED_RESULT" | python3 -c "import sys,json; d=json.load(sys.stdin); print(d['status'])")
-DETACHED_PATH=$(echo "$DETACHED_RESULT" | python3 -c "import sys,json; d=json.load(sys.stdin); print(d.get('worktree_path') or '')")
-DETACHED_REASON=$(echo "$DETACHED_RESULT" | python3 -c "import sys,json; d=json.load(sys.stdin); print(d.get('reason') or '')")
+DETACHED_PATH=""
+if [ "$DETACHED_FLAG" = "true" ]; then
+  # EC-15: prepare detached review once; compare runs reuse the same path.
+  DETACHED_JSON=$(python3 .map/scripts/map_step_runner.py prepare_detached_review)
+  DETACHED_PATH=$(echo "$DETACHED_JSON" | python3 -c "import sys,json; print(json.load(sys.stdin).get('detached_path',''))")
```
[the remark this answers is not in the document]
Comment (azalio) [reply .claude/skills/map-review/SKILL.md]: Fixed in a22ad14 in the .claude source copy; templates were resynced.

File: src/mapify_cli/templates/skills/map-review/SKILL.md [hunk 3]
```diff
@@ -196,336 +134,101 @@ git diff HEAD
...
-
-Task(
-  subagent_type="evaluator",
-  description="Score change quality",
-  prompt="[paste EVALUATOR_PROMPT]"
-)
+```text
+Task(subagent_type="monitor", description="Review diff for correctness", prompt="Read generated monitor prompt from build_review_prompts output.")
+Task(subagent_type="predictor", description="Predict integration risk", prompt="Read generated predictor prompt from build_review_prompts output.")
+Task(subagent_type="evaluator", description="Score review quality", prompt="Read generated evaluator prompt from build_review_prompts output.")
```
[the remark this answers is not in the document]
Comment (azalio) [reply src/mapify_cli/templates/skills/map-review/SKILL.md]: Fixed in b45c28e: build_review_prompts output is now parsed into MONITOR_PROMPT/PREDICTOR_PROMPT/EVALUATOR_PROMPT and those variables are passed to Task calls.

File: .claude/skills/map-review/SKILL.md [hunk 4]
```diff
@@ -196,336 +134,101 @@ git diff HEAD
...
-
-Task(
-  subagent_type="evaluator",
-  description="Score change quality",
-  prompt="[paste EVALUATOR_PROMPT]"
-)
+```text
+Task(subagent_type="monitor", description="Review diff for correctness", prompt="Read generated monitor prompt from build_review_prompts output.")
+Task(subagent_type="predictor", description="Predict integration risk", prompt="Read generated predictor prompt from build_review_prompts output.")
+Task(subagent_type="evaluator", description="Score review quality", prompt="Read generated evaluator prompt from build_review_prompts output.")
```
[the remark this answers is not in the document]
Comment (azalio) [reply .claude/skills/map-review/SKILL.md]: Fixed in b45c28e in the .claude source copy; templates were resynced.

File: src/mapify_cli/templates/skills/map-review/SKILL.md [hunk 5]
```diff
@@ -196,336 +134,101 @@ git diff HEAD
...
 
 ### Step B.0: Determine section presentation order
 
-Before presenting any section, call the ordering helper to get the section sequence.
-For default (no ordering flag), the helper returns canonical order. For `--reverse-sections`,
-`--shuffle-sections`, or compare-mode runs, the helper returns the chosen order.
-
 ```bash
-SECTIONS_JSON=$(python3 .map/scripts/map_step_runner.py shuffle-sections "$MODE_FLAG" "$SEED_RAW")
+SECTIONS_JSON=$(python3 .map/scripts/map_step_runner.py shuffle-sections --mode "$MODE_FLAG" --seed "$SEED_RAW")
```
[the remark this answers is not in the document]
Comment (azalio) [reply src/mapify_cli/templates/skills/map-review/SKILL.md]: Fixed in b45c28e: shuffle-sections now uses positional arguments again.

File: .claude/skills/map-review/SKILL.md [hunk 6]
```diff
@@ -196,336 +134,101 @@ git diff HEAD
...
 
 ### Step B.0: Determine section presentation order
 
-Before presenting any section, call the ordering helper to get the section sequence.
-For default (no ordering flag), the helper returns canonical order. For `--reverse-sections`,
-`--shuffle-sections`, or compare-mode runs, the helper returns the chosen order.
-
 ```bash
-SECTIONS_JSON=$(python3 .map/scripts/map_step_runner.py shuffle-sections "$MODE_FLAG" "$SEED_RAW")
+SECTIONS_JSON=$(python3 .map/scripts/map_step_runner.py shuffle-sections --mode "$MODE_FLAG" --seed "$SEED_RAW")
```
[the remark this answers is not in the document]
Comment (azalio) [reply .claude/skills/map-review/SKILL.md]: Fixed in b45c28e in the .claude source copy; templates were resynced.

File: src/mapify_cli/templates/skills/map-check/SKILL.md [hunk 7]
```diff
@@ -6,9 +6,10 @@ effort: low
 disable-model-invocation: true
 argument-hint: "[focus area]"
 ---
-# /map-check — Quality Gates & Verification
+# /map-check - Quality Gates & Verification
 
-**Purpose:** Run code quality checks (linters, type checkers, tests) and/or verify MAP workflow completion.
+Purpose: run verification only. Do not plan, implement, or fix from this skill.
```
[the remark this answers is not in the document]
Comment (azalio) [reply src/mapify_cli/templates/skills/map-check/SKILL.md]: Fixed in b45c28e: purpose text now says quality gates and MAP workflow verification.

File: .claude/skills/map-check/SKILL.md [hunk 8]
```diff
@@ -6,9 +6,10 @@ effort: low
 disable-model-invocation: true
 argument-hint: "[focus area]"
 ---
-# /map-check — Quality Gates & Verification
+# /map-check - Quality Gates & Verification
 
-**Purpose:** Run code quality checks (linters, type checkers, tests) and/or verify MAP workflow completion.
+Purpose: run verification only. Do not plan, implement, or fix from this skill.
```
[the remark this answers is not in the document]
Comment (azalio) [reply .claude/skills/map-check/SKILL.md]: Fixed in b45c28e in the .claude source copy; templates were resynced.

File: src/mapify_cli/templates/skills/map-check/SKILL.md [hunk 9]
```diff
@@ -27,254 +28,152 @@ parallel_tool_policy: independent_checks_only
...
-if [[ $? -ne 0 ]]; then
-  echo "❌ Tests failed - verification REJECTED"
-  VERDICT="REJECTED"
-  REASON="Test suite has failures"
-fi
+LOG_FILE=".map/${BRANCH}/tests.log"
+mkdir -p ".map/${BRANCH}"
+# Re-run the failing command into $LOG_FILE, then parse it:
+python3 .map/scripts/diagnostics.py parse --tool tests --log "$LOG_FILE" --command "$TEST_CMD" --exit-code $?
 ```
```
[the remark this answers is not in the document]
Comment (azalio) [reply src/mapify_cli/templates/skills/map-check/SKILL.md:171]: Fixed in b45c28e: the diagnostics snippet defines TEST_CMD, reruns it into LOG_FILE, captures TEST_EXIT, and passes that exit code to parse/summarize.

File: .claude/skills/map-check/SKILL.md [hunk 10]
```diff
@@ -27,254 +28,152 @@ parallel_tool_policy: independent_checks_only
...
-if [[ $? -ne 0 ]]; then
-  echo "❌ Tests failed - verification REJECTED"
-  VERDICT="REJECTED"
-  REASON="Test suite has failures"
-fi
+LOG_FILE=".map/${BRANCH}/tests.log"
+mkdir -p ".map/${BRANCH}"
+# Re-run the failing command into $LOG_FILE, then parse it:
+python3 .map/scripts/diagnostics.py parse --tool tests --log "$LOG_FILE" --command "$TEST_CMD" --exit-code $?
 ```
```
[the remark this answers is not in the document]
Comment (azalio) [reply .claude/skills/map-check/SKILL.md:171]: Fixed in b45c28e in the .claude source copy; templates were resynced.
